Encore Networking Patch
=======================

This patch modifies the standard libraries roundTrip function on the HTTP interface, such that Encore's
`encoreBeginRoundTrip` and `encoreFinishRoundTrip` are called. These calls are routed into Encore runtime and allows us
instrument the call.

diff --git a/src/net/http/transport.go b/src/net/http/transport.go
index 7561f7f5cb..302425e3b4 100644
--- a/src/net/http/transport.go
+++ b/src/net/http/transport.go
@@ -510,9 +510,17 @@ func (t *Transport) alternateRoundTripper(req *Request) RoundTripper {
 }

 // roundTrip implements a RoundTripper over HTTP.
-func (t *Transport) roundTrip(req *Request) (*Response, error) {
+func (t *Transport) roundTrip(req *Request) (rr *Response, err error) {
 	t.nextProtoOnce.Do(t.onceSetNextProtoDefaults)
-	ctx := req.Context()
+	ctx, err := encoreBeginRoundTrip(req)
+	if err != nil {
+		req.closeBody()
+		return nil, err
+	}
+	req.ctx = ctx
+	defer func() {
+		encoreFinishRoundTrip(req, rr, err)
+	}()
 	trace := httptrace.ContextClientTrace(ctx)

 	if req.URL == nil {
